File: fs/numbers/test/NumbersIterativeTest.fs
namespace test

open NUnit.Framework
open src.NumbersIterative

[<TestFixture>]
module IterativeTests =

    [<Test>]
    let ``Test Fibonacci iterative`` () =
        Assert.AreEqual(5, fibonacci 5)

    [<Test>]
    let ``Test Factorial iterative`` () =
        Assert.AreEqual(120, factorial 5)

    [<Test>]
    let ``Test sum_numbers iterative`` () =
        Assert.AreEqual(15, sum_numbers 5)

File: _experimental/_c++/numbers/test/numbers_iterative_test.cpp
#include <gtest/gtest.h>
#include "../src/interfaces/numbers_iterative.h"

class IterativeTest : public ::testing::Test
{
protected:
    NumbersIterative numbers;
};

TEST_F(IterativeTest, Fibonacci)
{
    EXPECT_EQ(numbers.fibonacci(5), 5);
}

TEST_F(IterativeTest, Factorial)
{
    EXPECT_EQ(numbers.factorial(5), 120);
}

TEST_F(IterativeTest, SumNumbers)
{
    EXPECT_EQ(numbers.sum_numbers(5), 15);
}